File: src/main/java/org/antop/java/chapter2/ReadNumber.java
package org.antop.java.chapter2;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;

public class ReadNumber {

    public static Integer readNumber(BufferedReader reader) throws IOException {
        try (reader) {
            var line = reader.readLine();
            return (line != null)
                    ? Integer.parseInt(line)
                    : null;
        } catch (NumberFormatException e) {
            return null;
        }
    }

    public static void main(String[] args) throws IOException {
        var reader = new BufferedReader(new StringReader(""));
        var n = readNumber(reader);
        System.out.println("number = " + n);
    }

}

File: src/main/kotlin/org/antop/kotlin/chapter2/ReadNumber.kt
package org.antop.kotlin.chapter2

import java.io.BufferedReader
import java.io.StringReader

fun readNumber(reader: BufferedReader) = try {
    reader.use {
        reader.readLine()?.toInt()
    }
} catch (e: NumberFormatException) {
    null
}

fun main() {
    val reader = BufferedReader(StringReader("11f"))
    val n = readNumber(reader)
    println("number = $n")
}
